Conversions/DateToDay.js

```diff
@@ -9,54 +9,43 @@
     algorithm shown below gives us the number of the day and
     finally converts it to the name of the day.
 
-    Algorithm & Explanation : https://cs.uwaterloo.ca/~alopez-o/math-faq/node73.html
+    Algorithm & Explanation : https://en.wikipedia.org/wiki/Zeller%27s_congruence
 */
 
-// March is taken as the first month of the year.
-const calcMonthList = {
-  1: 11,
-  2: 12,
-  3: 1,
-  4: 2,
-  5: 3,
-  6: 4,
-  7: 5,
-  8: 6,
-  9: 7,
-  10: 8,
-  11: 9,
-  12: 10
-}
-
-// show the week day in a number : Sunday - Saturday => 0 - 6
-const daysNameList = { // weeks-day
-  0: 'Sunday',
-  1: 'Monday',
-  2: 'Tuesday',
-  3: 'Wednesday',
-  4: 'Thursday',
-  5: 'Friday',
-  6: 'Saturday'
-}
+// Array holding name of the day: Saturday - Sunday - Friday => 0 - 1 - 6
+const daysNameArr = ['Saturday', 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday']
 
 const DateToDay = (date) => {
   // firstly, check that input is a string or not.
   if (typeof date !== 'string') {
     return new TypeError('Argument is not a string.')
   }
   // extract the date
-  const [day, month, year] = date.split('/').map((x) => Number(x))
+  let [day, month, year] = date.split('/').map((x) => Number(x))
   // check the data are valid or not.
-  if (day < 0 || day > 31 || month > 12 || month < 0) {
+  if (day < 1 || day > 31 || month > 12 || month < 1) {
     return new TypeError('Date is not valid.')
   }
+
+  // In case of Jan and Feb:
+  // Year: we consider it as previous year
+  // e.g., 1/1/1987 here year is 1986 (-1)
+  // Month: we consider value as 13 & 14 respectively
+  if (month < 3) {
+    year--
+    month += 12
+  }
+
   // divide year to century and yearDigit value.
   const yearDigit = (year % 100)
   const century = Math.floor(year / 100)
+
   // Apply the algorithm shown above
```

Comment:
Please add a note explaining that Java's modulo operator requires an adaption of the formula.

Reply:
Done. Updated comment